test-framework/junit5/src/main/java/io/quarkus/test/junit/util/QuarkusTestProfileAwareClassOrderer.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.test.junit.util;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.ClassDescriptor;
+import org.junit.jupiter.api.ClassOrderer;
+import org.junit.jupiter.api.ClassOrdererContext;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+
+public class QuarkusTestProfileAwareClassOrderer implements ClassOrderer {
+
+    @Override
+    public void orderClasses(ClassOrdererContext context) {
+        AtomicInteger testProfileCounter = new AtomicInteger();
+        Map<Class<? extends QuarkusTestProfile>, Integer> testProfileNumberMap = new HashMap<>();
+        Function<Class<? extends QuarkusTestProfile>, Integer> testProfileNumberProvider = desc -> testProfileNumberMap
```

Comment:
I don't really like the counter approach, as I would prefer something stable between runs. 

Ideally this would match the continuous testing sort order (ideally with a shared implementation): https://github.com/quarkusio/quarkus/blob/9950848f3ac7dfdf6b862431afabe2442b967bc4/core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java#L584

Reply:
I think it's stable between runs because e.g. surefire will not change the order randomly unless you tell it to do so.

But I will have a look at what you suggested.

Reply:
If it is stable then I think the approach is probably fine.

Reply:
I aligned it a bit more to what you suggested (profile FQCN infix), but I don't really see much value in trying to unify it more since the constraints are different: The order has a single list will all tests, the runner in continuous testing has two separate ones. I also wanted to give users the choice to reconfigure the order (or even subclass it).

Reply:
Btw, I quickly tried this with continuous testing and users might be surprised to see a different ordering there.
But I think we can always follow up with further unifications later.